lib/matplotlib/animation.py

```diff
@@ -557,6 +557,7 @@ def isAvailable(cls):
     def __init__(self, *args, **kwargs):
```

Comment:
`__init__(self, *args, loop=False, **kwargs)`
or at least make it a boolean.

Argument needs to be documented in some docstring.

Please check for API consistency across writers; if the behavior of other writers is to loop then the default should be changed here.

Reply:
The Pillow image.save function takes the ```loop``` parameter as an integer, so a gif could be make to loop a set number of times. Also, ```loop=False``` might make sense in the docstring, but ```loop=self.loop``` where ```self.loop``` is ```False``` is an endless loop since ```False==0```. I suppose I could change it to ```loop=not self.loop```, but should we not allow setting to loop say 5 times? 

I'll change it to a Boolean if you still want. (I suppose no one needs a gif to loop exactly 137 times!)

I like the move to the ```loop``` kwarg in the ```__init__``` function definition. I'll do that for sure.

HTMLWriter and ImageMagick both default to looping I believe, but the ffmpegwriter writing to a .mp4 wouldn't loop. Whether defaulting to loop or not really depends on the format. Gif's should loop, mp4's can't. Different formats have different looping properties (HTMLWriter has a 'reflect' option, so it has a ```default_mode``` parameter that takes a string instead of a ```loop``` parameter. 

I'll default it to looping (unless you object).

I know that I need to put it in a docstring, but I'm having difficulty figuring out where. I'm not familiar with sphinx-autodoc. I really could use help here. I could write an entire new docstring based off the base class and add the parameter. Or I could do as @jklymak suggested, but then other writers would have a "loop parameter only available in PillowWriter".